ibeam/src/two_fa_handlers/google_msg_handler.py

```diff
@@ -59,14 +60,18 @@ def get_two_fa_code(self) -> Union[str, None]:
 
             WebDriverWait(driver_2fa, 120).until(sms_code_present)
 
-        sms_list_el = driver_2fa.find_elements_by_class_name(_GOOG_MESSAGES_LIST_CLASS)
+        sms_list_el = driver_2fa.find_elements_by_css_selector(_GOOG_MESSAGES_LIST_CLASS)
 
         if not sms_list_el:
             _LOGGER.error('Timeout or authentication error while loading sms messages.')
         else:
             _LOGGER.info(sms_list_el[0].text)
+            sms_list_el[0].click() #mark message as read
+            time.sleep(2) #wait for click to mark message as read
```

Comment:
Should we maybe do some selenium `.until` waiting here? I'm not entirely sure, just a suggesting to make this more robust in case 2 seconds isn't enough.

Reply:
Ya I almost did but I couldn't immediately determine if the message is marked as read, I guessed at when location bar is updated or when the message pane loads, but it might be an asynchronous call unrelated to those and I could monitor the nav bar going from class unread to read. Actually that gives me an idea we could just check the element we all ready have's parent div (.text-content) to see if the unread class went away. But honestly, I highly doubt this code would ever fail as-is anyway.

Reply:
Wanna give it a shot then?
>  I highly doubt this code would ever fail as-is anyway.

Does the next line where we read the sms rely in some way on that message being marked as read?

Reply:
No, currently I'm just firing off a click() event to trigger the process of the web app to mark the message as read, and the sleep is only there to ensure the instance/tab/chromedriver isn't closed before the xhr is fired.

Reply:
I will update though, just not tonight and I don't think we need to wait for me to do it cause it's not going to hurt in the short term.

Reply:
Ahh gotcha 👍 Fair enough, I guess 2 seconds are reasonable then. Cool, I think it's ready to merge then. Well done :)